modules/EvseV2G/v2g_ctx.cpp

```diff
@@ -492,6 +482,20 @@ bool add_service_to_service_list(struct v2g_context* v2g_ctx, const struct iso2_
     return true;
 }
 
+void check_and_remove_service_in_list(struct v2g_context* v2g_ctx, uint16_t service_id) {
+
+    size_t id{0};
+    auto service_found{false};
+
+    for (auto service_it = v2g_ctx->evse_v2g_data.evse_service_list.begin();
+         service_it != v2g_ctx->evse_v2g_data.evse_service_list.end(); ++service_it) {
+        if (service_it->ServiceID == service_id) {
+            service_it = v2g_ctx->evse_v2g_data.evse_service_list.erase(service_it);
+            break;
```

Comment:
This could probably be done with some C++11-ism such as `v.erase(std::remove_if())`
```
    auto& v = v2g_ctx->evse_v2g_data.evse_service_list; // reference as abbreviation
    v.erase(std::remove_if(v.begin(), v.end(), [service_id](const auto list) { return list->ServiceID == service_id; }), v.end());
```
(Sorry, written blindly and untestet.)